packages/journey-manager/src/index.ts

```diff
@@ -226,12 +234,14 @@ export const run = (props: RunProps): RunOutput => {
   }
 
   const sendStep = (stepId: string, once: boolean): void => {
-    if (triggeredSteps.includes(stepId)) {
+    if (
+      triggeredSteps.some((triggeredStep) => triggeredStep.stepId === stepId)
+    ) {
       if (once) {
```

Comment:
_perf:_ probably would be more performant to do:

```javascript
if (once && triggeredSteps.some((triggeredStep) => triggeredStep.stepId === stepId))
```

(i.e. avoid checking the array if we didn't call it with `once`.